src/models/room.ts

```diff
@@ -2307,6 +2307,15 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
             receipts: this.cachedThreadReadReceipts.get(threadId) ?? [],
         });
 
+        // Add the re-emitter before we start adding events to the thread so we don't miss events
+        this.reEmitter.reEmit(thread, [
+            ThreadEvent.Delete,
+            ThreadEvent.Update,
+            ThreadEvent.NewReply,
+            RoomEvent.Timeline,
+            RoomEvent.TimelineReset,
+        ]);
+
```

Comment:
This fixes the notification bug (and is also generally much saner as far as I'm concerned): add the re-emitter straight away so it doesn't miss (node) events, rather then after we add (matrix) events.